controllers/vspherecluster_reconciler.go

```diff
@@ -256,11 +262,9 @@ func (r clusterReconciler) reconcileIdentitySecret(ctx *context.ClusterContext)
 		}
 
 		// check if cluster is already an owner
-		if !clusterutilv1.IsOwnedByObject(secret, vsphereCluster) {
-			if len(secret.GetOwnerReferences()) > 0 {
-				return fmt.Errorf("another cluster has set the OwnerRef for secret: %s/%s", secret.Namespace, secret.Name)
```

Comment:
Well I think we need to have the check that this secret is owned by a particular vSphereCluster object only? If this secret owned by another VSphereCluster, then we should throw an error.